graphql-dotnet/conventions
# Add workflows, samples, tests, and formatting checks

Overview of changes:
- Run `dotnet format` and apply all recommended changes
- Added workflows based on those used in graphql-dotnet/server, and removed old workflow
  - Test code when creating/updating a PR
  - Check formatting when creating/updating a PR
  - Fail if PR title contains '[WIP]'
  - Build code and upload nuget preview package to GitHub Packages when merging to master/develop
  - Publish to nuget.org when creating a release
- Added nuget secret to repository secrets
- Added readme and workflows to solution
- Added sample projects to solution, so they build during tests (todo: add tests)
- Remove outdated dependencies on GraphQL.SystemReactive and replace with System.Reactive
- Update subscription sample to use latest version of GraphQL.Server; update code and references as necessary
- Update SimpleWebApp to support Newtonsoft.Json (add `AllowSynchronousIO = true`)
- Add Playground UI package to SimpleWebApp and DataLoaderWithEFCore so they can be tested (neither works however)
- Update all GraphQL dependencies to 7.2.0
- Remove unnecessary dependencies from GraphQL.Conventions
- Update GraphQL.Conventions version to 7.1.0
- Added test project for SubscriptionExample, which verifies both POST and WebSocket subscription operations
- Add `schema.Initialize()` to 3 of 4 SchemaConstructorTests

No functional changes to GraphQL.Conventions were made in this PR, although some unnecessary/obsolete dependencies were removed.  Users may need to add a reference to System.Reactive 5.0.0.

Appveyor integrations may be removed with the merging of this PR.

Subsequent PRs might include these changes:
- Add Directory.build.props
- Add Tests.props
- Bump to .NET 7
- Add editorconfig
- Update csproj properties
- Remove global.json

Possible future changes:
- Remove or deprecate 'server'-type functionality; instead rely on the server project
- Add configuration extensions to `IGraphQLBuilder`
- Remove GraphQL.DataLoader dependency (maintaining support for data loaders)
- Replace `IDependencyInjector` with `IServiceProvider`
